ga/htdocs/components/90_ga.js

```diff
@@ -210,51 +276,191 @@ Ensembl.GA = {
 };
 
 /**
- * Prototype for eventConfigs
+ * Google Analytics 4 implementation
  */
-Ensembl.GA.EventConfig = function (config) {
-  $.extend(this, config);
-};
+ 
+Ensembl.GA4 = {
+  
+  domainCodes: {},
+  /*
+  * key pair value for domain name and it's corresponding code (specified later in plugins)
+  */
+  
+  code: function () {
+    /*
+     * Returns the ga code for the current domain
+     */
+    return this.domainCodes[window.location.hostname];
+  },
 
-Ensembl.GA.EventConfig.prototype.getURL = function (url) {
-  return Ensembl.GA.filterURL(url || this.currentTarget || this.currentOptions.url);
-};
+  init: function () {
+    /*
+     * Initialises google analytics
+     */
+    if (this.code() && !this.initialised) {
 
-Ensembl.GA.EventConfig.prototype.getText = function (el) {
-  return $(el || this.currentTarget).text().trim();
-};
+      var s = document.createElement("script");
+      s.type = "text/javascript";
+      s.src = "https://www.googletagmanager.com/gtag/js?id=" + this.code();
+      s.setAttribute('async', '');
+      $("head").append(s);
 
-Ensembl.GA.EventConfig.prototype.instantiate = function (extra, defaults, e) {
-  var config  = $.extend({}, defaults, this, extra);
-  config.data = $.extend({}, this.data);
+      // get the species regexp before calling filterURL
+      this.urlSpeciesRegex = new RegExp('/(' + Ensembl.allSpeciesList.join('|') + ')/');
```

Comment:
This is repeated in both GA4 and GAU. Better move it to GA where it is actually used?